management/src/clusterctl/main.go

```diff
@@ -179,7 +196,11 @@ func newPostActioner(validateCb validateCallback, postCb postCallback) *postActi
 }
 
 func (npa *postActioner) procFlags(c *cli.Context) {
-	npa.extraVars = c.String("extra-vars")
+	npa.flags.extraVars = c.String("extra-vars")
+
+	// c.String returns "" if the flag does not exist
```

Comment:
this is because the `default` value of the flag was set as empty string. It will return the default when user didn't mention the flag. We may want to update this comment accordingly or I would perhaps remove it altogether as it is the usual package behavior.